fix(DDS): fix dds task lint

zhangtao505 opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #fix dds task lint

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->

./scripts/codecheck.sh ./huaweicloud/services/dds

==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        8      3547      341        52     3154        485           108.28
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~ds/resource_huaweicloud_dds_instance_v3.go      1108       87        10     1011        220            21.76
~ces/dds/resource_huaweicloud_dds_backup.go       550       62        12      476         87            18.28
~/resource_huaweicloud_dds_database_role.go       381       35         3      343         42            12.24
~/resource_huaweicloud_dds_database_user.go       330       30         3      297         33            11.11
~s/data_source_huaweicloud_dds_instances.go       428       35         6      387         33             8.53
~urce_huaweicloud_dds_parameter_template.go       346       37         9      300         26             8.67
~/data_source_huaweicloud_dds_flavors_v3.go       149       19         0      130         23            17.69
~source_huaweicloud_dds_audit_log_policy.go       255       36         9      210         21            10.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     8      3547      341        52     3154        485           108.28
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 106510 bytes, 0.107 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
30 dds resourceDdsInstanceV3Update huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:568:1
17 dds flavorSpecCodeUpdate huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:1016:1
14 dds flavorSizeUpdate huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:948:1
13 dds resourceDdsInstanceV3Create huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:357:1
13 dds resourceDdsBackupCreate huaweicloud/services/dds/resource_huaweicloud_dds_backup.go:144:1
11 dds resourceDdsBackupDelete huaweicloud/services/dds/resource_huaweicloud_dds_backup.go:459:1
10 dds flavorNumUpdate huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:884:1
9 dds resourceDdsInstanceV3Read huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:450:1
9 dds ddsJobStatusRefreshFunc huaweicloud/services/dds/resource_huaweicloud_dds_backup.go:301:1
8 dds dataSourceDDSFlavorV3Read huaweicloud/services/dds/data_source_huaweicloud_dds_flavors_v3.go:77:1
Average: 4.17

==> Checking for golangci-lint...
huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:568:1: cyclomatic complexity 30 of func `resourceDdsInstanceV3Update` is high (> 20) (gocyclo)
func resourceDdsInstanceV3Update(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
^

==> Checking for Nolint directives...
./huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go:90:     time.Sleep(2 * time.Second) // lintignore:R018
./huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go:111:    time.Sleep(2 * time.Second) // lintignore:R018
./huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go:244:    time.Sleep(2 * time.Second) // lintignore:R018

==> Checking for TF features in dds...

==> Checking for misspell in dds...

==> Checking for code complexity in ./huaweicloud/services/acceptance/dds...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        8      1805      120         3     1682         96            51.62
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~source_huaweicloud_dds_instance_v3_test.go       972       40         0      932         37             3.97
~ds/resource_huaweicloud_dds_backup_test.go       139       16         1      122         19            15.57
~urce_huaweicloud_dds_database_user_test.go       120       11         0      109         14            12.84
~urce_huaweicloud_dds_database_role_test.go       162       12         0      150         14             9.33
~e_huaweicloud_dds_audit_log_policy_test.go       136       16         1      119          8             6.72
~huaweicloud_dds_parameter_template_test.go       140       13         1      126          4             3.17
~_source_huaweicloud_dds_flavors_v3_test.go        38        5         0       33          0             0.00
~a_source_huaweicloud_dds_instances_test.go        98        7         0       91          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     8      1805      120         3     1682         96            51.62
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 52287 bytes, 0.052 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
18 dds testAccCheckDDSV3InstanceFlavor huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:297:1
6 dds testAccDatabaseUserImportStateIdFunc huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_database_user_test.go:79:1
6 dds testAccDatabaseRoleImportStateIdFunc huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_database_role_test.go:77:1
5 dds getDdsResourceFunc huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:17:1
5 dds getDdsBackupResourceFunc huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_backup_test.go:18:1
Average: 2.18

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/dds...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/dds...

==> Cleanup patch...

Check Completed!



## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

 make testacc TEST=./huaweicloud/services/acceptance/dds TESTARGS='-run TestAccDDSV3Instance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds -v -run TestAccDDSV3Instance_basic -timeout 360m -parallel 4
=== RUN   TestAccDDSV3Instance_basic
=== PAUSE TestAccDDSV3Instance_basic
=== CONT  TestAccDDSV3Instance_basic
--- PASS: TestAccDDSV3Instance_basic (3270.04s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds       3270.079s

zhangtao505 committed:
./scripts/codecheck.sh ./huaweicloud/services/dds

==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        8      3547      341        52     3154        485           108.28
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~ds/resource_huaweicloud_dds_instance_v3.go      1108       87        10     1011        220            21.76
~ces/dds/resource_huaweicloud_dds_backup.go       550       62        12      476         87            18.28
~/resource_huaweicloud_dds_database_role.go       381       35         3      343         42            12.24
~/resource_huaweicloud_dds_database_user.go       330       30         3      297         33            11.11
~s/data_source_huaweicloud_dds_instances.go       428       35         6      387         33             8.53
~urce_huaweicloud_dds_parameter_template.go       346       37         9      300         26             8.67
~/data_source_huaweicloud_dds_flavors_v3.go       149       19         0      130         23            17.69
~source_huaweicloud_dds_audit_log_policy.go       255       36         9      210         21            10.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     8      3547      341        52     3154        485           108.28
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 106510 bytes, 0.107 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
30 dds resourceDdsInstanceV3Update huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:568:1
17 dds flavorSpecCodeUpdate huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:1016:1
14 dds flavorSizeUpdate huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:948:1
13 dds resourceDdsInstanceV3Create huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:357:1
13 dds resourceDdsBackupCreate huaweicloud/services/dds/resource_huaweicloud_dds_backup.go:144:1
11 dds resourceDdsBackupDelete huaweicloud/services/dds/resource_huaweicloud_dds_backup.go:459:1
10 dds flavorNumUpdate huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:884:1
9 dds resourceDdsInstanceV3Read huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:450:1
9 dds ddsJobStatusRefreshFunc huaweicloud/services/dds/resource_huaweicloud_dds_backup.go:301:1
8 dds dataSourceDDSFlavorV3Read huaweicloud/services/dds/data_source_huaweicloud_dds_flavors_v3.go:77:1
Average: 4.17

==> Checking for golangci-lint...
huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:568:1: cyclomatic complexity 30 of func `resourceDdsInstanceV3Update` is high (> 20) (gocyclo)
func resourceDdsInstanceV3Update(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
^

==> Checking for Nolint directives...
./huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go:90:     time.Sleep(2 * time.Second) // lintignore:R018
./huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go:111:    time.Sleep(2 * time.Second) // lintignore:R018
./huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go:244:    time.Sleep(2 * time.Second) // lintignore:R018

==> Checking for TF features in dds...

==> Checking for misspell in dds...

==> Checking for code complexity in ./huaweicloud/services/acceptance/dds...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        8      1805      120         3     1682         96            51.62
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~source_huaweicloud_dds_instance_v3_test.go       972       40         0      932         37             3.97
~ds/resource_huaweicloud_dds_backup_test.go       139       16         1      122         19            15.57
~urce_huaweicloud_dds_database_user_test.go       120       11         0      109         14            12.84
~urce_huaweicloud_dds_database_role_test.go       162       12         0      150         14             9.33
~e_huaweicloud_dds_audit_log_policy_test.go       136       16         1      119          8             6.72
~huaweicloud_dds_parameter_template_test.go       140       13         1      126          4             3.17
~_source_huaweicloud_dds_flavors_v3_test.go        38        5         0       33          0             0.00
~a_source_huaweicloud_dds_instances_test.go        98        7         0       91          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     8      1805      120         3     1682         96            51.62
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 52287 bytes, 0.052 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
18 dds testAccCheckDDSV3InstanceFlavor huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:297:1
6 dds testAccDatabaseUserImportStateIdFunc huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_database_user_test.go:79:1
6 dds testAccDatabaseRoleImportStateIdFunc huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_database_role_test.go:77:1
5 dds getDdsResourceFunc huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:17:1
5 dds getDdsBackupResourceFunc huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_backup_test.go:18:1
Average: 2.18

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/dds...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/dds...

==> Cleanup patch...

Check Completed!

ShiChangkuo commented on huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:
```diff
@@ -4,20 +4,20 @@ import (
 	client, err := c.DdsV3Client(acceptance.HW_REGION_NAME)
 	if err != nil {
-		return nil, fmtp.Errorf("Error creating HuaweiCloud DDS client: %s ", err)
+		return nil, fmt.Errorf("Error creating HuaweiCloud DDS client: %s ", err)
```
1. please delete HuaweiCloud
2. an extra space an the end of this line

ShiChangkuo commented on huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:
```diff
@@ -27,18 +27,17 @@ func getDdsResourceFunc(c *config.Config, state *terraform.ResourceState) (inter
-	if instances.TotalCount == 0 {
-		return nil, fmtp.Errorf("dds instance not found.")
+	if instanceList.TotalCount == 0 {
+		return nil, fmt.Errorf("dds instance not found ")
```
ditto

ShiChangkuo commented on huaweicloud/services/dds/data_source_huaweicloud_dds_flavors_v3.go:
```diff
@@ -75,25 +75,25 @@ func DataSourceDDSFlavorV3() *schema.Resource {
-	config := meta.(*config.Config)
-	ddsClient, err := config.DdsV3Client(config.GetRegion(d))
+	conf := meta.(*config.Config)
+	ddsClient, err := conf.DdsV3Client(conf.GetRegion(d))
```
it's better to define a variable: region := conf.GetRegion(d)

ShiChangkuo commented on huaweicloud/services/dds/resource_huaweicloud_dds_database_role.go:
```diff
@@ -226,15 +228,15 @@ func flattenDatabaseRoles(roleList []roles.RoleDetail) []map[string]interface{}
 
-func flattenDatabasePrivilegeResource(resource roles.Resource) []map[string]interface{} {
-	if (resource == roles.Resource{}) {
+func flattenDatabasePrivilegeResource(resources roles.Resource) []map[string]interface{} {
```
how about use roleResource or raw?

zhangtao505 committed:
make testacc TEST=./huaweicloud/services/acceptance/dds TESTARGS='-run TestAccDDSV3Instance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds -v -run TestAccDDSV3Instance_basic -timeout 360m -parallel 4
=== RUN   TestAccDDSV3Instance_basic
=== PAUSE TestAccDDSV3Instance_basic
=== CONT  TestAccDDSV3Instance_basic
--- PASS: TestAccDDSV3Instance_basic (3332.34s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds       3332.377s

ShiChangkuo commented on huaweicloud/services/dds/data_source_huaweicloud_dds_flavors_v3.go:
```diff
@@ -115,20 +116,20 @@ func dataSourceDDSFlavorV3Read(_ context.Context, d *schema.ResourceData, meta i
 	}
 
-	logp.Printf("Extract %d/%d flavors by filters.", len(flavorList), len(allFlavors))
+	log.Printf("Extract %d/%d flavors by filters.", len(flavorList), len(allFlavors))
```
logging level is missing

```suggestion
	log.Printf("[DEBUG] extract %d/%d flavors by filters.", len(flavorList), len(allFlavors))
```

ShiChangkuo commented on huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:
```diff
@@ -298,38 +297,38 @@ func TestAccDDSV3Instance_withConfigurationSingle(t *testing.T) {
+				if group.Type == "mongos" {
+					if len(group.Nodes) != v.(int) {
+						return fmt.Errorf(
+							"Error updating HuaweiCloud DDS instance: num of mongos nodes expect %d, but got %d",
```
suggest to replace all **HuaweiCloud**

zhangtao505 committed:
fix task lint
